tests/setupconfig/test_auth.py

```diff
@@ -0,0 +1,185 @@
+    assert config.oidc_rp_sign_algo == "HS256"
+    assert config.oidc_rp_idp_sign_key == ""
+    assert config.oidc_op_discovery_endpoint == ""
+    assert config.oidc_op_jwks_endpoint == ""
+
+    assert (
+        config.oidc_op_authorization_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/auth"
+    )
+    assert (
+        config.oidc_op_token_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/token"
+    )
+    assert (
+        config.oidc_op_user_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/userinfo"
+    )
+    assert config.username_claim == ["sub"]
+    assert config.groups_claim == ["roles"]
+    assert config.claim_mapping == {
+        "last_name": ["family_name"],
+        "first_name": ["given_name"],
+        "email": ["email"],
+    }
+    assert config.sync_groups
+    assert config.sync_groups_glob_pattern == "*"
+    assert config.default_groups.all().count() == 0
+    assert not config.make_users_staff
+    assert config.superuser_group_names == []
+    assert config.oidc_use_nonce
+    assert config.oidc_nonce_size == 32
+    assert config.oidc_state_size == 32
+    assert (
+        config.userinfo_claims_source == UserInformationClaimsSources.userinfo_endpoint
+    )
+
+
+@pytest.mark.vcr
+@pytest.mark.django_db
+def test_configure_use_discovery_endpoint(setup_config_discovery):
+    call_command("setup_configuration")
+
+    config = OpenIDConnectConfig.get_solo()
+
+    assert config.enabled
+    assert config.oidc_op_discovery_endpoint == KEYCLOAK_BASE_URL
+    assert (
+        config.oidc_op_jwks_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/certs"
+    )
+    assert (
+        config.oidc_op_authorization_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/auth"
+    )
+    assert (
+        config.oidc_op_token_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/token"
+    )
+    assert (
+        config.oidc_op_user_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/userinfo"
+    )
+
+
+@pytest.mark.django_db
+def test_configure_discovery_failure(requests_mock, setup_config_discovery):
+    mock_kwargs = (
+        {"exc": requests.ConnectTimeout},
+        {"exc": requests.ConnectionError},
+        {"status_code": 404},
+        {"status_code": 403},
+        {"status_code": 500},
+    )
+    for mock_config in mock_kwargs:
+        requests_mock.get(
+            f"{KEYCLOAK_BASE_URL}.well-known/openid-configuration",
+            **mock_config,
+        )
+
+        with pytest.raises(ConfigurationRunFailed):
+            AdminOIDCConfigurationStep().configure()
+
+        assert not OpenIDConnectConfig.get_solo().enabled
+
+
+@pytest.mark.django_db
+def test_is_configured(setup_config_full):
+    config = AdminOIDCConfigurationStep()
+
+    assert not config.is_configured()
+
+    config.configure()
+
+    assert config.is_configured()
```

Comment:
Can you explain what this actually means/implies?

Reply:
IDK tbh. Copied it from open-iwoner

Reply:
The purpose is that if it is configured correctly, it should return configured. So if it is already configure, the setup command can fix it.

Not sure this test really says anything though. is_configured just returns the enabled attribute

Reply:
This feature is now removed and so is the test